tests/opentracer/test_tracer.py

```diff
@@ -32,12 +33,9 @@ def test_config(self):
         assert tracer._enabled is True
 
     def test_no_service_name(self):
-        """Config without a service_name should raise an exception."""
-        from ddtrace.settings import ConfigException
-
-        with pytest.raises(ConfigException):
-            tracer = Tracer()
-            assert tracer is not None
+        """A service_name should be generated if one is not provided."""
+        tracer = Tracer()
+        assert tracer._service_name
```

Comment:
Check the value. We need to ensure that if we change the logic, the value is not changed (so the software is not changed).